### PR TITLE
feat(orchestration): add synthesis phase to spectacular workflow

### PR DESCRIPTION
**Pull request by jsboigeEpita**
## Summary
- Add terminal `synthesis` phase to spectacular workflow, depends_on quality/counter/debate/governance/formal_synthesis
- Aggregates all analysis outputs into a structured synthesis report
- Spectacular workflow: 20 → 21 phases

## Changes
- **workflows.py**: New phase `synthesis` with capability `analysis_synthesis`
- **invoke_callables.py**: `_invoke_analysis_synthesis` collecting state fields into sections
- **state_writers.py**: `_write_analysis_synthesis_to_state` populating `state.analysis_synthesis`
- **registry_setup.py**: `analysis_synthesis_service` registration
- **5 tests**: Phase presence, DAG deps, state writer, registry, synthesis output

## Test plan
- [x] `pytest tests/unit/argumentation_analysis/orchestration/test_synthesis_spectacular.py -v` — 5 passed
- [ ] CI green
- [ ] Verify spectacular state ends with populated `analysis_synthesis` field

Closes #508

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: argumentation_analysis/orchestration/invoke_callables.py [hunk 1]
```diff
@@ -3791,6 +3803,106 @@ def _count_referenced_fields(state: Any) -> int:
...
+        if formal and isinstance(formal, (list, dict)):
+            sections[field_name] = {"present": True, "size": len(formal)}
+
+    # Overall assessment
+    total_phases = len(phase_results)
+    sections_count = len(sections)
+    overall_completeness = (
+        sections_count / max(total_phases, 1) if total_phases > 0 else 0.0
+    )
+
```
Comment (clusterManager-Myia) [inline argumentation_analysis/orchestration/invoke_callables.py:3896]: Nit: `max(total_phases, 1)` is redundant here — the ternary already guards `total_phases > 0`, so `total_phases >= 1` on the true branch. Simplify to:

```python
overall_completeness = (
    sections_count / total_phases if total_phases > 0 else 0.0
)
```

Same result, less noise.

File: argumentation_analysis/orchestration/invoke_callables.py [hunk 2]
```diff
@@ -3791,6 +3803,106 @@ def _count_referenced_fields(state: Any) -> int:
...
+    governance, and formal phases (from context), and produces a
+    structured summary with scores and key findings.
+    """
+    state = context.get("_state_object")
+    phase_results = {}
+
+    # Collect phase outputs from context
+    for key, val in context.items():
+        if (
+            key.startswith("phase_")
```
Comment (clusterManager-Myia) [inline argumentation_analysis/orchestration/invoke_callables.py:3829]: The ternary branches for `isinstance(quality_scores, list)` and `else` both compute `len(quality_scores)` — the conditional is a no-op. Just use `len(quality_scores)` directly.

Same pattern appears below for `jtms_beliefs`.

File: argumentation_analysis/orchestration/invoke_callables.py [hunk 3]
```diff
@@ -2799,21 +2800,17 @@ async def _invoke_asp_reasoning(
     # Try JVM + Tweety ClingoSolver first
     try:
         from argumentation_analysis.core.jvm_setup import is_jvm_started
```
Comment (clusterManager-Myia) [inline argumentation_analysis/orchestration/invoke_callables.py:2805]: This PR mixes feature additions with unrelated line-length reformatting in `_invoke_asp_reasoning`, `_invoke_modal_logic`, and `build_formal_extended_workflow`. Consider separating formatting commits into a standalone PR or at least a distinct commit to keep the feature diff minimal and reviewable.

File: tests/unit/argumentation_analysis/orchestration/test_synthesis_spectacular.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
+"""Tests for synthesis phase wiring in spectacular (#508)."""
```
Comment (clusterManager-Myia) [inline tests/unit/argumentation_analysis/orchestration/test_synthesis_spectacular.py:1]: Consider adding a unit test for `_invoke_analysis_synthesis` itself (e.g., with a mock state object that has various attributes populated, verifying the output structure and completeness calculation). The current tests cover wiring well but leave the callable's aggregation logic untested — especially edge cases like empty state, partial state, or missing phase outputs.

**Review by clusterManager-Myia — APPROVED** [review]
✅ **Approved** — clean architecture, follows existing patterns, CI green.

**What's good:**
- Correct DAG wiring: synthesis correctly depends on quality/counter/debate/governance/formal_synthesis (transitively covers jtms/atms/dung)
- Phase count 20→21 verified by test
- State writer properly guards with `hasattr` before writing
- Registry registration follows the established dynamic-class pattern
- 5 wiring tests covering phase presence, deps, state writer, registry, and output

**Suggestions (non-blocking, see inline comments):**
1. **Dead ternary** in `_invoke_analysis_synthesis` — `isinstance(quality_scores, list) ? len(...) : len(...)` both branches identical (same for `jtms_beliefs`). Just use `len()` directly.
2. **Redundant `max()`** in completeness calc — guarded by `total_phases > 0` already.
3. **Mixed formatting changes** — the PR bundles unrelated line-length reformatting in `_invoke_asp_reasoning`, `_invoke_modal_logic`, and `build_formal_extended_workflow`. Consider isolating these in a separate commit for cleaner history.
4. **Test the callable** — wiring tests are solid but `_invoke_analysis_synthesis`'s aggregation logic has no direct test (empty state, partial state, edge cases).

None of these are blockers. Nice addition to the spectacular workflow. 🚀

**Review by clusterManager-Myia — APPROVED** [review]
Approved by Hermes Agent after review. 4 non-blocking suggestions left as inline comments (dead ternary, redundant max(), mixed formatting, missing callable-level tests). Architecture is sound, CI green, ready to merge.
